lib/api/redfish-1.0/update-service.js

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+
+'use strict';
+
+var injector = require('../../../index.js').injector;
+var redfish = injector.get('Http.Api.Services.Redfish');
+var Promise = injector.get('Promise'); // jshint ignore:line
+var _ = injector.get('_');  // jshint ignore:line
+var controller = injector.get('Http.Services.Swagger').controller;
+var Errors = injector.get('Errors');
+var workflow = injector.get('Http.Services.Api.Workflows');
+var waterline = injector.get('Services.Waterline');
+
+var updateFirmware = controller({success: 201}, function (req, res) {
+    var payload = req.swagger.params.payload.value;
+    var targetNodeIds = payload.Targets;
+    var errorList = [];
+    var transactionError = false;
+    var options = redfish.makeOptions(req, res, payload);
+
+    return Promise.try(function () {
+        // Run update firmware graph for each specified node id
+        // If there are any errors, the entire request fails
+        return Promise.map(
+            targetNodeIds,
+            function (targetNodeId) {
+                return waterline.nodes.getNodeById(targetNodeId)
+                    .then(function (targetNode) {
+                        if (!targetNode) {
+                            transactionError = true;
+                            throw new Errors.NotFoundError(
+                                'Node not Found ' + targetNodeId);
+                        }
+                    })
+                    .then(function () {
+                        return waterline.obms.findAllByNode(targetNodeId, true);
+                    })
+                    .then(function (obmSettings) {
+                        if (!obmSettings) {
+                            transactionError = true;
+                            throw new Errors.NotFoundError(
+                                'OBM settings not defined for node: ' + targetNodeId);
+                        }
+
+                        // Find IPMI settings for the node
+                        var obm = _.find(obmSettings, {'service': 'ipmi-obm-service'});
+                        if (!obm) {
+                            transactionError = true;
+                            throw new Errors.NotFoundError(
+                                'IPMI settings not defined for node: ' + targetNodeId);
+                        }
+
+                        if (!transactionError) {
+                            return workflow.createAndRunGraph(
+                                {
+                                    name: 'Graph.Dell.Racadm.Update.Firmware',
```

Comment:
Since the API entry `/redfish/v1/UpdateService/Actions/UpdateService.SimpleUpdate` is common for all kinds of servers, so this controller function should take all servers into consideration rather than only Dell.  But for your design, it is actually only valid for Dell server.

We certainly have defined different firmware update graphs for different servers, but they are not unlike Dell (one graph for all firmwares), those server defines different graph for different firmware, so there is certainly some challenge to support all servers.

This comment is not push you to support all servers, but as a reminder for you. In the point of feature integrity, it deserves some following action items.